R/facet-labels.r

```diff
@@ -1,137 +1,264 @@
-#' Label facets with their value.
-#' This is the default labelling scheme.
+#' Labeller functions
 #'
-#' @param variable variable name passed in by facetter
-#' @param value variable value passed in by facetter
-#' @family facet labellers
-#' @export
+#' Labeller functions are in charge of formatting the strip labels of
+#' facet grids and wraps. Most of them accept a \code{multi_line}
+#' argument to control whether multiple factors (defined in formulae
+#' such as \code{~first + second}) should be displayed on a single
+#' line separated with commas, or each on their own line.
+#'
+#' \code{label_value()} only displays the value of a factor while
+#' \code{label_both()} displays both the variable name and the factor
+#' value. \code{label_context()} is context-dependent and uses
+#' \code{label_value()} for single factor facetting and
+#' \code{label_both()} when multiple factors are
+#' involved. \code{label_wrap_gen()} uses \code{\link[base]{strwrap}()}
+#' for line wrapping.
+#'
+#' \code{\link{label_parsed}()} interprets the labels as plotmath
+#' expressions. \code{\link{label_bquote}()} offers a more flexible
+#' way of constructing plotmath expressions. See examples and
+#' \code{\link{bquote}()} for details on the syntax of the
+#' argument.
+#'
+#' @section Writing New Labeller Functions:
+#'
+#'   A labeller function accepts a dataframe of labels (character
+#'   vectors) containing one column for each factor. Muultiple factors
+#'   occur with formula of the type \code{~first + second}.
+#'
+#'   The return value must be a dataframe or a rectangular list of
```

Comment:
I tried to find a way to express that clearly, but the problem with data frames is that the returned list can be unnamed or have names that are not meaningful. One additional subtlety is that the character vectors can be lists of plotmath expressions as well, so if we talk about data frames we need to be talking about list-columns which is complicated as well.

I reworded the paragraph in the last commit with an illustration to make the idea of a rectangular list clearer (hopefully).